src/Nethermind/Nethermind.Trie/BatchedTrieVisitor.cs

```diff
@@ -290,14 +290,12 @@ void QueueNextNodes(ArrayPoolList<(TrieNode, TNodeContext, SmallTrieVisitContext
         Interlocked.Decrement(ref _activeJobs);
     }
 
-
     private void BatchedThread()
     {
-        using ArrayPoolList<(TrieNode, TNodeContext, SmallTrieVisitContext)> nextToProcesses = new(_maxBatchSize);
-        using ArrayPoolList<int> resolveOrdering = new(_maxBatchSize);
-        ArrayPoolList<(TrieNode, TNodeContext, SmallTrieVisitContext)>? currentBatch;
+        using ArrayPoolList<(TrieNode, TNodeContext, SmallTrieVisitContext)> nextToProcesses = new(_maxBatchSize); // I tried to change to ArrayPoolListRef but tests are failing, dunno why
```

Comment:
@asdacap could you take a look why this fails as struct ref?
I see this can be invoked many times, would be nice saving...